toot/api.py

```diff
@@ -289,7 +289,19 @@ def _notification_timeline_generator(app, user, path, params=None):
     while path:
         response = http.get(app, user, path, params)
         notification = response.json()
-        yield [n["status"] for n in notification if n["status"]]
+
+        statuses = []
+        for n in notification:
+            status = n["status"]
+            # hack to have notification info available in our statuses
+            # so we don't have to redo all interfaces.
+            # preface with toot_ so we don't collide
+            # with real status tags now or in future
+            status.setdefault('toot_notification_id', n.get('id'))
+            status.setdefault('toot_notification_type', n.get('type'))
+            status.setdefault('toot_notification_created_at', n.get('created_at'))
```

Comment:
Hm, I dislike this hack, since it's polluting the `api` module with things that only concern the TUI. I think we should hold off on this until I get a chance to finish #313, then we can put this in `StatusMeta`.